services/121-service/src/utils/send-xlsx-response.ts

```diff
@@ -1,5 +1,11 @@
 import { Response } from 'express';
 import * as XLSX from 'xlsx';
+import * as zlib from 'zlib';
+
+// Use compression and chunks to write into xls
+// for each data sheet of xlsx file to avoid memory error
+const CHUNK_SIZE = 300000;
```

Comment:
Hey, I was wondering about this. Does this CHUNK_SIZE corresponds to the number of rows? If so than is this chunk_size of 300.000 not too large since we only support programs of up to 100.000. Or do I misunderstand this @ashishkumarbasak-accenture

Reply:
Do you know how long an export now takes with 1.5 million rows?

Reply:
@RubenGeo This is chunk size to write into excel file data sheet, where each data sheet will contain 300,000 rows. So if we have 700,000 rows, then it will create 3 data sheets in the excel file.

I have tested with chunk size 500,000 rows and it works fine as well, but to be on safe side I made it a little smaller value, 300,000. If you want I can make it to 500,000 as chunk size, but small value than 300,000 will help use to write the file, but in that case the excel file will have many sheets.

Performance wise on my local machine it took around ~ 4 min to complete.